src/Microsoft.TemplateEngine.Utils/InMemoryFileSystem.cs

```diff
@@ -499,7 +499,7 @@ public IEnumerable<string> EnumerateFileSystemEntries(string directoryName, stri
                 }
             }
 
-            Regex rx = new Regex(Regex.Escape(pattern).Replace("\\*", ".*").Replace("\\?", "."));
+            Regex rx = new Regex("^" + Regex.Escape(pattern).Replace("\\*", ".*").Replace("\\?", ".") + "$");
```

Comment:
Change makes sense. But, do we want to do the replaces on the pattern and then escape the replaced value, as opposed to escaping the pattern and then doing the replacements?